### PR TITLE
Feature/fm 406 detailed search about the work

### PR DESCRIPTION
**Pull request by ManFromMons**
This first check-in is to ensure the build works and colleagues are happy with the additions.

### CODE REVIEW

File: app/models/facilities_management/procurement_building.rb [hunk 1]
```diff
@@ -32,6 +33,21 @@ def cleanup_service_codes
       self.service_codes = service_codes.reject(&:blank?)
     end
 
+    def services_valid?
+      false if procurement_building_services.empty?
+
+      result = procurement_building_services.all? { |pbs| pbs.valid?(:all) }
+      errors.add(:procurement_building_services, :invalid, message: 'Some services are invalid') unless result
+
+      # unless result
```
[the remark this answers is not in the document]
Comment (ManFromMons) [reply app/models/facilities_management/procurement_building.rb]: Done

File: app/models/facilities_management/procurement_building_service.rb [hunk 2]
```diff
@@ -26,75 +26,127 @@ class ProcurementBuildingService < ApplicationRecord
...
+    def validate_services
+      located_services = SERVICES_AND_QUESTIONS.select { |x| x[:code] == code }
+
+      results = {}
+      error_collection = ActiveModel::Errors.new(ProcurementBuildingService)
+
+      validate_volume
+      error_collection.merge!(errors)
+
+      located_services.each do |svc|
```
[the remark this answers is not in the document]
Comment (ManFromMons) [reply app/models/facilities_management/procurement_building_service.rb]: Done

File: app/helpers/facilities_management/beta/procurements_helper.rb [hunk 3]
```diff
@@ -82,4 +82,10 @@ def extension_four_description
   def any_service_codes(procurement_buildings)
     procurement_buildings.map(&:service_codes).flatten.reject(&:blank?).any?
   end
+
+  def building_services_status?(status)
+    return t('common.Incomplete') unless status == true
```
[the remark this answers is not in the document]
Comment (ManFromMons) [reply app/helpers/facilities_management/beta/procurements_helper.rb]: ? where

[the remark this answers is not in the document]
Comment (kennyevil) [reply app/helpers/facilities_management/beta/procurements_helper.rb]: The translation keys. Please don't capitalise them.

**Pull request merged**